bindings_ffi/src/mls.rs

```diff
@@ -2633,127 +2578,134 @@ pub fn decode_reaction(bytes: Vec<u8>) -> Result<FfiReaction, GenericError> {
         .map_err(|e| GenericError::Generic { err: e.to_string() })
 }
 
-#[derive(uniffi::Enum, Clone, Default, PartialEq, Debug)]
-pub enum FfiReactionAction {
-    Unknown,
-    #[default]
-    Added,
-    Removed,
-}
+// RemoteAttachmentInfo and MultiRemoteAttachment FFI structures - using types from message module
 
-impl From<FfiReactionAction> for i32 {
-    fn from(action: FfiReactionAction) -> Self {
-        match action {
-            FfiReactionAction::Unknown => 0,
-            FfiReactionAction::Added => 1,
-            FfiReactionAction::Removed => 2,
-        }
-    }
-}
+#[uniffi::export]
+pub fn encode_multi_remote_attachment(
+    ffi_multi_remote_attachment: FfiMultiRemoteAttachment,
+) -> Result<Vec<u8>, GenericError> {
+    // Convert FfiMultiRemoteAttachment to MultiRemoteAttachment
+    let multi_remote_attachment: MultiRemoteAttachment = ffi_multi_remote_attachment.into();
 
-#[derive(uniffi::Enum, Clone, Default, PartialEq, Debug)]
-pub enum FfiReactionSchema {
-    Unknown,
-    #[default]
-    Unicode,
-    Shortcode,
-    Custom,
+    // Use MultiRemoteAttachmentCodec to encode the reaction
+    let encoded = MultiRemoteAttachmentCodec::encode(multi_remote_attachment)
+        .map_err(|e| GenericError::Generic { err: e.to_string() })?;
+
+    // Encode the EncodedContent to bytes
+    let mut buf = Vec::new();
+    encoded
+        .encode(&mut buf)
+        .map_err(|e| GenericError::Generic { err: e.to_string() })?;
+
+    Ok(buf)
 }
 
-impl From<FfiReactionSchema> for i32 {
-    fn from(schema: FfiReactionSchema) -> Self {
-        match schema {
-            FfiReactionSchema::Unknown => 0,
-            FfiReactionSchema::Unicode => 1,
-            FfiReactionSchema::Shortcode => 2,
-            FfiReactionSchema::Custom => 3,
-        }
-    }
+#[uniffi::export]
+pub fn decode_multi_remote_attachment(
+    bytes: Vec<u8>,
+) -> Result<FfiMultiRemoteAttachment, GenericError> {
```

Comment:
Could be misunderstanding the overall intent - for this and the other methods, is the decode method needed by the native SDK's, or are we decoding these automatically in Rust when messages are queried?

If decoding in rust, why not have the encode also happen automatically in Rust, by passing the `FfiMultiRemoteAttachment` directly to send API's? Then we could remove these methods and shrink the libxmtp<>SDK interface